blockprod/src/rpc.rs

```diff
@@ -38,6 +54,37 @@ impl BlockProductionRpcServer for super::BlockProductionHandle {
     async fn start(&self) -> rpc::Result<()> {
         handle_error(self.call(|this| this.start()).await)
     }
+
+    async fn generate_block(
+        &self,
+        reward_destination_hex: String,
+        transactions_hex: Vec<String>,
+    ) -> rpc::Result<Id<Block>> {
+        let reward_destination_data =
+            hex::decode(reward_destination_hex).map_err(rpc::Error::to_call_error)?;
+
+        let reward_destination = Destination::decode(&mut &reward_destination_data[..])
+            .map_err(rpc::Error::to_call_error)?;
+
+        let mut signed_transactions = vec![];
+
+        for transaction_hex in transactions_hex {
```

Comment:
Please use map, then collect to `Result<Transaction, Error>` then use the try operator `?`.